jujugui/static/gui/src/app/assets/javascripts/react-click-outside.js

```diff
@@ -21,10 +21,22 @@ function enhanceWithClickOutside(WrappedComponent) {
     componentDidMount: function componentDidMount() {
       this.__wrappedComponent = this.refs.wrappedComponent;
       document.addEventListener('click', this.handleClickOutside, true);
+      document.addEventListener('keydown', this.handleKeyDown);
     },
 
     componentWillUnmount: function componentWillUnmount() {
       document.removeEventListener('click', this.handleClickOutside, true);
+      document.removeEventListener('keydown', this.handleKeyDown);
+    },
+
+    handleKeyDown: function handleKeyDown(e) {
+      e = e || window.event;
+      if (e.keyCode == 27) {
+        var domNode = ReactDOM.findDOMNode(this);
+        if (domNode && typeof this.refs.wrappedComponent.handleClickOutside === 'function') {
```

Comment:
Might be worth a quick comment as to why you have to check the type of the function.